src/classes/OPP_StageMappingCMT.cls

```diff
@@ -0,0 +1,193 @@
+*/
+public with sharing class OPP_StageMappingCMT {
+    private static final String COMMITTED_STATE_VAL = 'COMMITTED';
+    private static final String FINALIZED_STATE_VAL = 'FINALIZED';
+    private static final String OPP_STAGE_TO_STATE_METADATA_NAME = 'Opportunity_Stage_To_State_Mapping__mdt';
+    private static final String OPP_STAGE_FIELD = 'Opportunity_Stage__c';
+    private static final String OPP_STATE_FIELD = 'Opportunity_State__c';
+    private static final String PLEDGED_STAGE = 'PLEDGED';
+
+    public OPP_StageMappingCMT() {
+
+    }
+
+    /*******************************************************************************************************
+    * @description Creates the finalized state record if there are no finalized records. It utilizes the 
+    * "Opportunity Stage when Fully Paid" field as the Opportunity stage for the finalized state. 
+    ********************************************************************************************************/
+    public static List<Metadata.CustomMetadata> createFinalizedOppStageToStateRecord(List<Metadata.CustomMetadata> metadataToInsert) {
+        npe01__Contacts_And_Orgs_Settings__c contactAndOrgCustomSetting = UTIL_CustomSEttingsFacade.getOrgContactsSettings();
+
+        // Only create the finalized opp stage to state record if there's no existing finalized record. 
+        Integer finalizedCount = [SELECT Count() FROM Opportunity_Stage_To_State_Mapping__mdt Where Opportunity_State__c =: FINALIZED_STATE_VAL];
+        
+        // Take the value, if there is one, and create a finalized Opp Stage to State custom metadata type record. 
+        if (finalizedCount == 0 && 
+            (contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != null && 
+            contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != '')) {
+            
+            metadataToInsert.add(createCustomMetadata(contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c, FINALIZED_STATE_VAL));
+        }
+
+        return metadataToInsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates the initial Opportunity Stage to State record if there are no existing records. 
+    * When there are no existing records, it will create a new "Committed" state record for a "Pledged" stage
+    * if there is one. 
+    ********************************************************************************************************/
+    public static List<Metadata.CustomMetadata> createPledgedOppStageToStateRecord(List<Metadata.CustomMetadata> metadataToInsert) {
+        npe01__Contacts_And_Orgs_Settings__c contactAndOrgCustomSetting = UTIL_CustomSEttingsFacade.getOrgContactsSettings();
+
+        
+        Integer overallCount = [SELECT Count() FROM Opportunity_Stage_To_State_Mapping__mdt];
+
```

Comment:
Why not return here if a record is present, rather than iterating over the the picklist entries?  Also...since we are using this only to tell whether or not there is one or more records present, I usually do a 

select Id from <<Object>>> limit 1, and then check for 1 or zero records.  It is more performant than a count()